File: src/main/java/com/example/coupon/controller/CouponCreate.java
class CouponCreateTest : BehaviorSpec({

    given("a CouponCreate record") {
        val name = "TestCoupon"
        val price = 100
        val count = 1000L

        `when`("the record is created") {
            val couponCreate = com.example.coupon.controller.CouponCreate(name, price, count)

            then("the values should be correctly set") {
                couponCreate.name shouldBe name
                couponCreate.price shouldBe price
                couponCreate.count shouldBe count
            }
        }
    }
})

File: src/test/kotlin/com/example/coupon/controller/CouponCreate.kt
class CouponCreateTest : FunSpec({
    test("CouponCreate record can be instantiated") {
        // Given:
        val name = "TestCoupon"
        val price = 100
        val count = 10L

        // When:
        val couponCreate = CouponCreate(name, price, count)

        // Then:
        couponCreate.name shouldBe name
        couponCreate.price shouldBe price
        couponCreate.count shouldBe count
    }
})